src/Factory/SchemaFromConveyorFactory.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Yiisoft\Yii\Cycle\Factory;
+
+use Closure;
+use Cycle\ORM\Schema;
+use Cycle\Schema\Compiler;
+use Cycle\Schema\GeneratorInterface;
+use Cycle\Schema\Registry;
+use Psr\Container\ContainerInterface;
+use Psr\SimpleCache\CacheInterface;
+use Spiral\Database\DatabaseManager;
+use Yiisoft\Yii\Cycle\Conveyor\SchemaConveyorInterface;
+
+final class SchemaFromConveyorFactory
+{
+    public string $cacheKey;
+    public bool $cacheEnabled;
+    /** @var GeneratorInterface[]|string[]|Closure[] */
+    private array $generators;
+
+    public function __construct(
+        bool $cacheEnabled = true,
+        string $cacheKey = 'Cycle-ORM-Schema',
+        array $generators = []
+    ) {
+        $this->cacheEnabled = $cacheEnabled;
+        $this->cacheKey = $cacheKey;
+        $this->generators = $generators;
+    }
+
+    /**
+     * @param ContainerInterface $container
+     * @return Schema
+     * @throws \Psr\SimpleCache\InvalidArgumentException
+     * @throws \Yiisoft\Yii\Cycle\Exception\BadGeneratorDeclarationException
+     */
+    public function __invoke(ContainerInterface $container)
+    {
+        $schemaArray = null;
+        if ($this->cacheEnabled) {
+            $container->get(CacheInterface::class)->get($this->cacheKey);
```

Comment:
probably should be like that?

`$schemaArray = $container->get(CacheInterface::class)->get($this->cacheKey);`